general/mem_manager.hpp

```diff
@@ -600,6 +600,12 @@ class MemoryManager
                          MemoryType h_mt, MemoryType d_mt,
                          bool own, bool alias, unsigned &flags);
 
+   /** @brief If true, class Memory will register the base Memory when creating
+       aliases with Memory<T>::MakeAlias(). */
```

Comment:
The other condition is that `!(base.flags & REGISTERED)`

Reply:
Yes, this should be clear though -- if the base is already registered, the method does not need to register it again.